blog-posts/emails/article.md

```diff
@@ -0,0 +1,329 @@
+
+Let's say I want to add a logo in my email's header. Using templated emails, it will be impossible to have it displayed on every mail client. The solution is to send a raw email with your image as attachment, and to reference it in the message's body, using contentID.
+
+I modified my email's header MJML code to include an image:
+
+```xml
+<mj-section background-color="#F48668" border-radius="8px 8px 0 0" padding="0">
+  <mj-column>
+    <mj-image width="20px" src="cid:my-logo@pchol.fr" alt="My logo"/>
+  </mj-column>
+</mj-section>
+```
+
+Then, in the lambda I use to send my emails, I can use the Nodemailer library to translate my MJML template into a raw email, and to add attachments to my message.
+
+```js
+import * as aws from '@aws-sdk/client-ses';
+import nodemailer from 'nodemailer';
+import myHtmlTemplate from './myHtmlTemplate'; // imported as a string
+
+export const handler = async () => {
+  const sesClient = new aws.SESClient({});
+  const transporter = nodemailer.createTransport({
+    SES: { ses: sesClient, aws },
+  });
+
+  await transporter.sendEmail({
+    from: 'notifications@pchol.fr', // my domain name
+    to: '<recipient>',
+    subject: 'This a raw email test',
+    text: 'This a raw email test',
+    html: myHtmlTemplate.replace('{{username}}', 'Pchol').replace('{{password}}', 'very_secret_password'), // manually replace parameters
+    attachments: [
+      {
+        filename: 'pchol-logo.png',
+        path: '/opt/pchol-logo.png', // image stored in a Lambda layer
+        cid: 'my-logo@pchol.fr', // same as in the template
+      },
+    ],
+  });
+
+  return 'OK!';
+};
+```
+
+The files that are attached to my mails come from a Lambda layer. These layers can be easily set on the different available frameworks.
+
+Using additional parameters like `list` or `headers`, you can specify custom headers for your emails (which is impossible using templated emails). For instance, to add a `List-unsubscribe` header (the last obstacle to your mail-tester perfect score), you can add the following code to your handler:
+
+```typescript
+list: {
+  unsubscribe: {
+    url: 'http://www.pchol.fr',
+    comment: 'Unsubscribe from this notifications',
+  },
+},
+```
+
+Here is the final result, containing the logo as an attachment and the `List-unsubscribe` header. The full mail, with html styles, attachment and custom headers obtains a 100% score on mail-tester when I send it from my custom domain based SES identity.
+
+![Final email with attachment](./assets/email-final.png 'Final email with attachment')
+
+## Conclusion
+
+From handling AWS sender reputation, to defining a responsive UI, going through setting appropriate DNS records on your domain name, this article aims at covering most common mistakes made to implement SES email sending capabilities and basic countermeasures. These problems are often encountered while sending naive emails with SES, and make a difference between professional looking messaging and spams.
```

Comment:
As there are a lots of things you covered in your article, why not add a little summary of the things to check in order to be sure your emails arrive in your recipient maibox with a good design ?

Reply:
I agree, an opinionated list of things to check without going into details like you already did before would be appreciated I think
```
1- monitoring of reputation
2 - alerting
...
j - use mjml
...
n - test your email with mail-tester
...
```